eva_cttv_pipeline/clinvar_xml_io/clinvar_xml_io/repeat_variant.py

```diff
@@ -64,44 +66,56 @@ def parse_repeat_identifier(variant_name):
+        repeat_type = repeat_type_from_length(repeat_unit_length)
+        if not repeat_type:
+            repeat_type = repeat_type_from_length(coordinate_span)
+
+    # Check if the HGVS-like name of the variant contains a simple deletion. In this case, it should not be
+    # processed as a repeat *expansion* variant. The reason such records are present at this stage is that for
+    # records without explicit allele sequences we cannot verify whether they definitely represent expansions.
+    if identifier and (identifier.endswith('del') or identifier.endswith('del)')):
         repeat_type = None
 
-        if self.is_protein_hgvs:
-            # For protein HGVS notation, assume that repeat is a trinucleotide one, since it affects entire amino acids
-            repeat_type = 'trinucleotide_repeat_expansion'
-        else:
-            # As a priority, use the repeat unit length determined directly from the HGVS-like base sequence
-            # If not available, fall back to using and end coordinate difference
-            repeat_unit_length = self.repeat_unit_length
-            if repeat_unit_length is None:
-                repeat_unit_length = self.coordinate_span
-            # Determine repeat type based on repeat unit length
-            if repeat_unit_length is not None:
-                if repeat_unit_length % 3 == 0:
-                    repeat_type = 'trinucleotide_repeat_expansion'
-                else:
-                    repeat_type = 'short_tandem_repeat_expansion'
-        # Check if the HGVS-like name of the variant contains a simple deletion. In this case, it should not be
-        # processed as a repeat *expansion* variant. The reason such records are present at this stage is that for
-        # records without explicit allele sequences we cannot verify whether they definitely represent expansions.
-        if self.name and (self.name.endswith('del') or self.name.endswith('del)')):
-            repeat_type = None
-        return repeat_type
+    return repeat_type, transcript_id
+
+
+def parse_all_identifiers(clinvar_measure: ClinVarRecordMeasure):
+    """Attempt to parse all identifiers (HGVS expressions and names) in a measure to get repeat type and transcript id.
+    Repeat type must be present if possible, transcript id is useful if present."""
+    repeat_type, transcript_id = infer_repeat_type_and_transcript_id(clinvar_measure.get_variant_name_or_hgvs())
+    if repeat_type:
+        return repeat_type, transcript_id
+
+    # TODO should we prioritise these in some way?
+    for hgvs in clinvar_measure.current_hgvs:
+        repeat_type, transcript_id = infer_repeat_type_and_transcript_id(hgvs.text)
+        if repeat_type:
+            return repeat_type, transcript_id
+    for name in clinvar_measure.all_names:
+        repeat_type, transcript_id = infer_repeat_type_and_transcript_id(name)
+        if repeat_type:
+            return repeat_type, transcript_id
```

Comment:
Wouldn't it be easier/clearer to  chain all names in a list and have a single loop ?
```suggestion
    # TODO should we prioritise these in some way?
    all_names = [clinvar_measure.get_variant_name_or_hgvs()] +\
                [hgvs.text for hgvs in clinvar_measure.current_hgvs] +\
                clinvar_measure.all_names
    for name in all_names:
        repeat_type, transcript_id = infer_repeat_type_and_transcript_id(name)
        if repeat_type:
            return repeat_type, transcript_id
```